portfolio/src/main/java/com/google/sps/servlets/LikeServlet.java

```diff
@@ -0,0 +1,50 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/like")
+public class LikeServlet extends HttpServlet {
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {    
+    String likeComment = request.getParameter("like");
+    if (likeComment==null || likeComment.isEmpty()){
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Comment must be a non-empty string.");
+      return;
+    }
+
+    Filter eqContent = new FilterPredicate("content", FilterOperator.EQUAL, likeComment);
```

Comment:
Does this work if there are two identical comments on your website?

Reply:
Yes and when  `commentEntity = pq.asSingleEntity();` is called later, it should produce a `PreparedQuery.TooManyResultsException` and return. In that case neither comment get up-voted.

Reply:
I would call that behavior a bug. Any thoughts about ways to fix it?

Reply:
I have a few ways in mind: 

1. Since people can only give thumbs up to comments that are displayed on the site, we can always narrow the search with the `limit` they have selected. 
2. But the method above doesn't work when duplicate comments are shown in the drop-down menu. A way to get around that is to replace the value attribute of each option with the key of the comment entity. The key is always unique even though the content might not be. 
3. In this application, the owner of the comment is not recorded, so it makes more sense to me if we combine the votes of two comments of same content. This can be achieved by adding a search in `submitComment()`: submitting an existing comment is equivalent of giving the old one a thumbs up and updating the timestamp. As a result there won't be duplicates in the database. However this does modify the UI because submitting a new comment may or may not update the list on the website. 

Thoughts on which way I should take (or better solution)?

Reply:
I believe (2) is the best solution. In youtube and other sites, some kind of id is exposed in the front-end. For example, YouTube comments have an id attached to them. If you click on the comment timestamp, refreshes the page with the comment pinned to the top and if you look at the URL you see a query param lc=<linked_comment_id>. When someone makes an action on a comment, it sends that comment id to the back-end. So you can do the same.

Reply:
I didn't notice that about yt comments until today! Initially I was concerned about displaying the id to the front end but I guess it isn't that big an issue if the database itself isn't public. Will resolve this conversation when I complete (2).